stdlib/asyncio/tasks.pyi

```diff
@@ -289,9 +289,9 @@ class Task(Future[_T], Generic[_T]):
 if sys.version_info >= (3, 7):
     def all_tasks(loop: AbstractEventLoop | None = ...) -> set[Task[Any]]: ...
     if sys.version_info >= (3, 8):
-        def create_task(coro: Generator[_TaskYieldType, None, _T] | Awaitable[_T], *, name: str | None = ...) -> Task[_T]: ...
+        def create_task(coro: Generator[Any, None, _T] | Coroutine[Any, Any, _T], *, name: str | None = ...) -> Task[_T]: ...
     else:
-        def create_task(coro: Generator[_TaskYieldType, None, _T] | Awaitable[_T]) -> Task[_T]: ...
```

Comment:
Notice that `_TaskYieldType` was not used.

Reply:
This doesn't seem right. Event loops typically use yields to communicate between tasks and the loop, so the generator must yield objects of a specific type.

Reply:
Some experiments:

```python
import asyncio
def gen():
   i = yield
   print(1, i)  # prints: `1, None`
   return 'a'

async def main():
  print('res', await asyncio.create_task(gen()))  # prints: `res, a`

asyncio.run(main())
```

And:

```python
import asyncio
def gen():
   i = yield
   print(1, i)  # prints: `1, None`
   yield 'a'

async def main():
  print('res', await asyncio.create_task(gen()))

asyncio.run(main())
```

```
1 None
Traceback (most recent call last):
  File "/Users/sobolev/Desktop/cpython/build/ex.py", line 10, in <module>
    asyncio.run(main())
  File "/Users/sobolev/.pyenv/versions/3.9.9/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Users/sobolev/.pyenv/versions/3.9.9/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/Users/sobolev/Desktop/cpython/build/ex.py", line 8, in main
    print('res', await asyncio.create_task(gen()))  # prints: `res, a`
  File "/Users/sobolev/Desktop/cpython/build/ex.py", line 5, in gen
    yield 'a'
RuntimeError: Task got bad yield: 'a'
```